src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1495,6 +1497,161 @@ static bool extendedTypeIsOrDerivedFrom(NamedTypeSymbol hiddenContainer, NamedTy
             }
         }
 
+        /// <summary>
+        /// We remove non-methods that are "hidden" (based on extended type) by more specific members.
+        /// But we collect all the methods, because we'll only know if they are "hidden" after we have the complete set of extension members.
+        ///
+        /// Once we have the complete set of extension members, we'll call <see cref="CompleteMergeExtensionLookupResultsHidingLessSpecific(LookupResult, ConsList{TypeSymbol}, ref CompoundUseSiteInfo{AssemblySymbol})"/>
+        /// It will remove methods that are "hidden" by any remaining non-methods.
+        /// It will also compact the set (deal with any null entries left in the result).
+        /// </summary>
+        private void MergeExtensionLookupResultsHidingLessSpecificNonMethods(LookupResult current, LookupResult additional, ConsList<TypeSymbol> basesBeingResolved, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
```

Comment:
>MergeExtensionLookupResultsHidingLessSpecificNonMethods [](http://example.com/codeflow?start=21&length=55)

The logic in this method is different enough from ```MergeHidingLookupResults``` to be concerned about a possible behavior differences. I think, it would be better to use exactly the same algorithm. I would even prefer a copy/paste with small changes just where it really matters.

I would like to suggest trying the following. In ```LookupImplicitExtensionMembersInSingleBinder``` topologically sort ```compatibleExtensions``` by inheritance relationship between the underlying types (as we sort base interface). Of course, we should account for the fact that multiple extensions can extend the same underlying type. Then preform the lookup in the order of that sort. This will ensure that we always look at derived types first, eliminating the problem around enforcing symmetry during the merge.